### PR TITLE
planner: add extractor for tikv_region_peers

### PR DESCRIPTION
**Pull request by IcePigZDB**
Signed-off-by: IcePigZDB <icepigzdb@gmail.com>

### What problem does this PR solve?

Problem Summary:
    The optimization idea is to add `TikvRegionPeersExtractor`  to push down `STORE_ID` and `REGION_ID` and  add `tikvRegionPeersRetriever`  to fetch regions information from PD.

### What is changed and how it works?
* ref issue: #28330

* relative executor PR #30657

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by rleungx — APPROVED** [review]

File: planner/core/memtable_predicate_extractor_test.go [hunk 1]
```diff
@@ -1420,3 +1420,126 @@ func (s *extractorSuite) TestTiDBHotRegionsHistoryTableExtractor(c *C) {
 		}
 	}
 }
+
+func (s *extractorSuite) TestTikvRegionPeersExtractor(c *C) {
+	se, err := session.CreateSession4Test(s.store)
+	c.Assert(err, IsNil)
+
+	var cases = []struct {
```
Comment (JmPotato) [inline planner/core/memtable_predicate_extractor_test.go:1428]: What about adding a test case that contains other columns to make sure we can only get the `region_id` and `store_id`?

**Review by JmPotato — APPROVED** [review]

**Comment by IcePigZDB** [comment]
@Yisaer  CI pass now, bot seems to stop the merge process.

**Comment by IcePigZDB** [comment]
@JmPotato  It failed again, is there anything I need to change?

**Comment by JmPotato** [comment]
> @JmPotato It failed again, is there anything I need to change?

@IcePigZDB I think it's just that the CI is unstabilized. Let's keep retrying until they are past...

**Review by JmPotato — APPROVED** [review]

**Comment by JmPotato** [comment]
> tide Pending — Not mergeable. PullRequest is missing sufficient approving GitHub review(s)

It's weird...Let's find another reviewer to approve again.

**Review by nolouch — APPROVED** [review]

**Pull request merged**